libs/shared/ui/src/lib/components/button/button.tsx

```diff
@@ -1,76 +1,70 @@
-import { Color } from '../../enums/colors.enum'
 import Icon from '../icon/icon'
 import { IconEnum } from '../../enums/icon.enum'
+import { Link } from 'react-router-dom'
 
 export enum ButtonSize {
-  BIG = 'BIG',
-  NORMAL = 'NORMAL',
-  SMALL = 'SMALL',
+  BIG = 'big',
+  NORMAL = 'normal',
+  SMALL = 'small',
+  VERY_SMALL = 'very-small'
 }
 
 export enum ButtonType {
-  BASIC = 'BASIC',
-  RAISED = 'RAISED',
-  STROKED = 'STROKED',
-  FLAT = 'FLAT',
-  ICON = 'ICON',
-  FAB = 'FAB',
-  MINI_FAB = 'MINI_FAB',
+  BASIC = 'basic',
+  RAISED = 'raised',
+  STROKED = 'stroked',
+  FLAT = 'flat'
 }
 
-/* eslint-disable-next-line */
 export interface ButtonProps {
   children: React.ReactNode
   size?: ButtonSize
   type?: ButtonType
-  color?: Color
-  iconLeft?: IconEnum
-  iconRight?: IconEnum
+  iconLeft?: IconEnum | string
+  iconRight?: IconEnum | string
+  link?: string
+  disabled?: boolean
+  className?: string
 }
 
 export function Button(props: ButtonProps) {
   const {
     children,
     size = ButtonSize.NORMAL,
     type = ButtonType.FLAT,
-    color = Color.VIOLET,
     iconLeft,
     iconRight,
+    link = '/',
+    disabled = false,
+    className = ''
   } = props
 
-  const defineClassName = () => {
-    let className = ''
-
-    switch (size) {
-      case ButtonSize.NORMAL:
-        className += ` h-9 px-4`
-        break
-      case ButtonSize.BIG:
-        className += ` h-12 px-4`
-        break
-    }
-
-    switch (type) {
-      case ButtonType.FLAT:
-        className += ` bg-${color}-500 hover:bg-${color}-600 text-white`
-        break
-      case ButtonType.STROKED:
-        className += ` border border-${color}-500 text-${color}-500`
-        break
-    }
-
-    return className
+  function content() {
+    return (
+      <>
+        {iconLeft ? <Icon name={iconLeft} /> : null} 
```

Comment:
If you render a component you can add directly {iconLeft && <Icon name={iconLeft} />}, the `undefined` value on the DOM should not display